### PR TITLE
Authorizations list

### PR DESCRIPTION
**Pull request by ConstantinKV**
- replaced RoundImageView with RoundedBitmapTransformation  
- update AuthorizationsListFragment  
- create AuthorizationsListViewModel  
- removed some unused resources  

device-2020-05-05-153907  

device-2020-05-05-153947  

device-2020-05-05-162245  

device-2020-05-05-162604

### CODE REVIEW

File: app/src/main/kotlin/com/saltedge/authenticator/tools/GlideTools.kt [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
 /**
- * Loads image from remote resource to image view
+ * Loads image from remote resource to image view and transform to rounded bitmap
  *
  * @receiver image view
  * @param imageUrl - remote image url
  * @param placeholderId - image to be set as default
  */
-fun ImageView.loadImage(imageUrl: String?, placeholderId: Int) {
+fun ImageView.loadRoundedImage(imageUrl: String?, placeholderId: ResId, cornerRadius: Float = -1f) {
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/tools/GlideTools.kt:21]: -1f?

File: app/src/main/kotlin/com/saltedge/authenticator/widget/EmptyView.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        actionView?.setInvisible(text == null)
+        actionView?.isClickable = text != null
+        actionView?.text = text
+    }
+
+    fun setIconResource(@DrawableRes resId: Int) {
+        imageView?.setImageResource(resId)
+    }
+
+    override fun setOnClickListener(l: OnClickListener?) {
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/widget/EmptyView.kt:73]: Follow the hierarchy: override/public/private

File: app/src/main/res/drawable-night/ic_authorizations_empty.xml [hunk 3]
```diff
@@ -0,0 +1,135 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (mnewlive) [inline app/src/main/res/drawable-night/ic_authorizations_empty.xml:1]: Please add `Copyright`

Comment (mnewlive) [reply app/src/main/res/drawable-night/ic_authorizations_empty.xml:1]: Can we use it this file for both themes?

File: app/src/main/res/drawable/ic_status_denied.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="56dp"
+    android:height="56dp"
+    android:viewportWidth="56"
+    android:viewportHeight="56">
+  <path
+      android:pathData="M23.5053,25H32.4947L45.9489,25.6511C46.5386,25.6796 47,26.1043 47,26.6186V29.3814C47,29.8957 46.5386,30.3204 45.9489,30.3489L32.4947,31L23.5053,31L10.0511,30.3489C9.4614,30.3204 9,29.8957 9,29.3814L9,26.6186C9,26.1043 9.4614,25.6796 10.0511,25.6511L23.5053,25Z"
+      android:fillColor="#FA5B5B"/>
```
Comment (mnewlive) [inline app/src/main/res/drawable/ic_status_denied.xml:8]: Why don't you use a universal color for two themes?

Comment (mnewlive) [reply app/src/main/res/drawable/ic_status_denied.xml:8]: Don't forget about the license

File: app/src/main/res/layout/view_empty.xml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:paddingTop="@dimen/dp_60"
+    android:gravity="center_horizontal"
+    android:background="@color/theme_background"
+    android:orientation="vertical">
+
+    <ImageView
+        android:id="@+id/imageView"
```
Comment (mnewlive) [inline app/src/main/res/layout/view_empty.xml:27]: imageView  - not informative

File: app/src/main/kotlin/com/saltedge/authenticator/tools/GlideTools.kt (hunk 1 above)
Comment (ConstantinKV) [reply app/src/main/kotlin/com/saltedge/authenticator/tools/GlideTools.kt:21]: -1f mean that value is not set.

File: app/src/main/res/drawable-night/ic_authorizations_empty.xml (hunk 3 above)
Comment (ConstantinKV) [reply app/src/main/res/drawable-night/ic_authorizations_empty.xml:1]: in some places there are single resource for both themes

File: app/src/main/res/drawable/ic_status_denied.xml (hunk 4 above)
Comment (ConstantinKV) [reply app/src/main/res/drawable/ic_status_denied.xml:8]: in some places there are single resource for both themes

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt [hunk 6]
```diff
@@ -21,149 +21,130 @@
...
-        viewModels = viewModels.filter { !it.shouldBeDestroyed }
-        viewContract?.updateViewsContent()
+        val currentItems = listItemsValues.filter { !it.shouldBeDestroyed }
+        if (currentItems != listItemsValues) postListItemsUpdate(currentItems)
+    }
+
+    private fun postListItemsUpdate(currentItems: List<AuthorizationViewModel>) {
+        listItems.postValue(currentItems)
+        if (currentItems.isEmpty()) {
+            emptyViewVisibility.postValue(View.VISIBLE)
```
[the remark this answers is not in the document]
Comment (ConstantinKV) [reply app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt]: i have added dataBinding

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListFragment.kt [hunk 7]
```diff
@@ -65,29 +66,27 @@ class AuthorizationsListFragment : BaseFragment(), AuthorizationsListContract.Vi
...
-        contentAdapter?.listItemClickListener = presenter
+        contentAdapter?.listItemClickListener = viewModel
     }
 
     override fun onResume() {
         super.onResume()
-        presenter.onResume()
+        clearAllNotifications()
         headerAdapter?.startTimer()
+
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListFragment.kt:89]: remove space

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,233 @@
+/*
+ * This file is part of the Salt Edge Authenticator distribution
+ * (https://github.com/saltedge/sca-authenticator-android).
+ * Copyright (c) 2019 Salt Edge Inc.
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt:4]: 2020

**Pull request merged**